### DISCUSSION TITLE
Populating name from relationship table

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [Budibase/budibase#6360] [Help]
I have been searching the internet but cannot find an answer to this question...  

I have a job table, a customer table and a subcontractor table...
When I insert jobs, I link them to the relevant customer and/or subcontractor using the relationship column.
The problem this gives is that it is impossible to filter the job table by customer or subcontractor and for me this is something which is very important for invoicing purposes etc.

I have added another 2 columns into the table, one for the customer name and the other for the subcontractor name.
I then played about with the `onchange` function with the goal of having the 2 new columns populated with the customer/subcontractor name when I chose them from the relationship dropdown,

However, the column gets populated with something like `%5B1%5D` which is from the auto generated ID column.

I have been trying to figure out a command I can run where I can get it to populate the actual supplier/cutomer name rather than the "code" ID

Is there a way to achieve this?

I have thought about adding a JS script... if "relationship_dropdown" = x Return y...

But New customers/suppliers get added regularly and I would need to amend the JS essentially daily and it would involve editing it on a number of different forms so i would rather avoid this

**Comment by melohagan** [comment]
Hi @PaulMcF1987 

This is potentially a limitation of the OnChange for a picker.
The binding `{{ Field Value }}` refers to the value of the chosen option, which would correctly be the primary key that you mentioned e.g. `%5B1%5D`

There is no binding for the Label of the picker (which is the primary display value you see in the dropdown).

As you're using an external datasource, you should be able to filter on the foreign key column.

![Screenshot 2022-06-17 at 10 50 11](https://user-images.githubusercontent.com/101575380/174274378-9886646d-4ead-4451-8d6a-7a04285aeef8.png)

![Screenshot 2022-06-17 at 10 50 51](https://user-images.githubusercontent.com/101575380/174274490-f9f326b9-999b-4f48-a839-484dec325d76.png)

**Reply by PaulMcF1987** [reply]
@melohagan I think the primary key column may be filterable but that is only useful if you know the number associated with the column needed.
In a workplace like this where there are a number of employees it is not feasible to search on this basis.

Its not the filter that you have shown above if it was that then it wouldnt be as big an obsticle. It is for the `dynamic filter` to allow us to search entries

For example, if we do 100 jobs in a month and each job uses a different sub contractor, that is 100 different subbies...
Sometimes subcontractors dont invoice straight away so there can be months where 200+ invoices come in.
We then need to search for the jobs that these invoices relate to, and the most obvious place to start and most efficient way would be to search for the name of the company who has invoiced
To take time away to look through a table of 1000+ subcontractors to then filter and do this 200 times would take forever.

I have tried searching for JS functions that I could try etc but not been able to find any.

Is there a work around?

**Reply by PaulMcF1987** [reply]
@melohagan After a job is entered into the table... 
you can use a text block with the handlebars `{{ element.0.primaryDisplay }}` and it shows the `label` of the Entry.

Given that it is possible to do this, I would be very surprised if there was no way to do what I have asked above.
I appreciate that there might not be a straight forward solution.
Essentially I just need a script which I can add to the `onchange` section which queries the database and returns the column stated as opposed to the ID.
The ID which it returns `%5B1%5D` is not even a stored column, so in terms of the actual table it has no relevance. 

> There is no binding for the Label of the picker

Can I create a binding?
